third_party/amd/lib/TritonAMDGPUToLLVM/OptimizeLDSUtility.cpp

```diff
@@ -73,11 +73,10 @@ createTmpLayout(triton::gpu::DistributedEncodingTrait layout,
         src.getKWidth());
   }
   if (auto src = dyn_cast<triton::gpu::SliceEncodingAttr>(layout)) {
-    // TODO: think of a way to construct slice layouts based on warpsPerCTA
-    // argument
-    auto parentWarpsPerCTA = triton::gpu::getWarpsPerCTA(src.getParent());
+    auto warps = to_vector(warpsPerCTA);
+    warps.insert(warps.begin() + src.getDim(), 1);
```

Comment:
Is there a bug here where `prod(warps)` isn't guarunteed to equal `num_warps`?

Reply:
Why exactly?

I think there was  bug before where it was not using `warpsPerCTA` but the current approach looks reasonable to me, given that they want to create any temporary layout of that type.

I also think that there is something wrong in the design of this pass if they need this sort of helper, but that's a story for another time.

Reply:
nvm, I looked at the users and they all guaruntee`prod(warpPerCTA) == num_warps` so there is no issue.